### PR TITLE
Cloudedit

### PR DESCRIPTION
**Pull request by aheifetz**
fixed null

### CODE REVIEW

File: lib/ec2/Instance.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ Instance = Resource.registerPropertyPrototypes(Instance, propertyMap);
 // -- Special, additional property prototypes -- //
 ///////////////////////////////////////////////////
 Instance.prototype.publicIP = function(subnetId, GroupSet){
-    if( this.node.Properties.NetworkInterfaces === null){
+    if( this.node.Properties.NetworkInterfaces == null){
```
Comment (owap) [inline lib/ec2/Instance.js]: How about if we just say 
```javascript
if( !this.node.Properties.NetworkInterfaces )
```
instead?

File: lib/ec2/Instance.js [hunk 2]
```diff
@@ -100,4 +100,25 @@ Instance.prototype.userDataTemplateFile = function (path, data) {
     this.userData(userData);
 };
 
+///////////////////////////////////////////////////
+// -- ResourceOutputs -- //
+///////////////////////////////////////////////////
+Instance.prototype.GetPublicIP = function(){
```
Comment (owap) [inline lib/ec2/Instance.js]: Is there some kind of distinction to be made from the first letter being upper-case when the others are lower?

File: lib/ec2/Instance.js (hunk 1 above)
Comment (owap) [inline lib/ec2/Instance.js]: Also, can we add a unit test for catching this issue? I can't even run the unit tests unless JSHint doesn't pass happily

File: lib/ec2/Instance.js (hunk 2 above)
Comment (aheifetz) [inline lib/ec2/Instance.js]: It seems mixed - happy to adopt any conventions.  camel case or upper camel case.  Let me know what you want

File: lib/ec2/Instance.js (hunk 1 above)
Comment (aheifetz) [inline lib/ec2/Instance.js]: Sounds good :)  I've been "functional" test by running the output through AWS.  The unit tests pass then it breaks on AWS.  Maybe need integration level tests?

Comment (owap) [inline lib/ec2/Instance.js]: The unit tests run the templates through the validator, but I guess the SDK is not as good at validating as the builder.  We can develop an integration test that uses the builder online, but making sure it immediately tears down everything if the build is successful will be crucial to saving costs.

File: lib/ec2/Instance.js (hunk 2 above)
Comment (owap) [inline lib/ec2/Instance.js]: The vast majority of the javascript starts with a lowercase function names, but this is inconsistent with the names of the functions in the [Amazon Documentation](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-template-resource-type-ref.html).

How about if we leave this off for now, and I'll go through and change the case sensitivity in a massive pull request later today?

**Comment by aheifetz** [comment]
fixed some of the issues

File: lib/ec2/Instance.js [hunk 3]
```diff
@@ -78,6 +78,29 @@ Instance.prototype.name = function (name) {
...
+                            '\n',
+                            'echo \"ulimit -n 65536\n\" > /etc/sysconfig/docker',
+                            '\n',
+                            'cat>/etc/security/limits.conf<<EOL\n',
+                            'root soft nofile 65536\n',
+                            'root hard nofile 65536\n',
+                            'EOL\n',
+                            'ulimit -n 65536\n',
+                            'service docker start'
+         ] ] } } ;
```
Comment (owap) [inline lib/ec2/Instance.js]: This script assumes the use of a `yum`-based package manager; is that what we want?

File: lib/ec2/Instance.js [hunk 4]
```diff
@@ -94,6 +117,39 @@ Instance.prototype.userDataTemplateFile = function (path, data) {
     var userDataStr = '{"Fn::Base64": { "Fn::Join": ["", [' + lines2.join(',') + ']]}}';
     var userData = JSON.parse(userDataStr);
     this.userData(userData);
+    return this;
+};
+
+Instance.prototype.userDataTemplateFile = function (path, data) {
 };
```
Comment (owap) [inline lib/ec2/Instance.js]: Is this a placeholder for some other function?

File: lib/ec2/Instance.js (hunk 3 above)
Comment (aheifetz) [inline lib/ec2/Instance.js]: Good point - I'll send an email with what we're trying to accomplish.  A better approach is to sub-class instance with organizational standard convenience features!

File: lib/ec2/Instance.js (hunk 4 above)
Comment (aheifetz) [inline lib/ec2/Instance.js]: I didn't think I added that but will take it out :)

**Comment by owap** [comment]
Any objections to me closing this pull request?  Is there an issue we want to log with desired functionality to implement?

**Pull request closed without merging**